src/app/item-card/item-card.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, Input, Output, EventEmitter } from "@angular/core";
+import { Item } from "../model/item.interface";
+
+@Component({
+  selector: "app-item-card",
+  templateUrl: "./item-card.component.html",
+  styleUrls: ["./item-card.component.scss"],
+})
+export class ItemCardComponent {
+  private maxDescriptionLength: number = 150;
+  private maxTitleLength: number = 14;
+  public itemData!: Item;
+  public description!: string;
+  public title!: string;
+
+  @Input() set item(item: Item) {
+    this.itemData = item;
+
+    this.description =
+      item.description.length > this.maxDescriptionLength
+        ? this.limitDescription(item.description)
+        : item.description;
+
+    this.title =
+      item.title.length > this.maxTitleLength
+        ? item.title.substring(0, this.maxTitleLength) + "..."
+        : item.title;
+  }
+
+  @Output()
+  addToFavorite = new EventEmitter<Item>();
+
+  @Output()
+  removeFromFavorite = new EventEmitter<Item>();
+
+  public addFavorite(item: Item): void {
```

Comment:
Do you think you need to emit the whole item for this kind of logic?

Reply:
Just the id would be enough right

Reply:
Yes.